### PR TITLE
SOLR-15625 Improve documentation for the benchmark module.

### PR DESCRIPTION
**Pull request by markrmiller**
[SOLR-15625 Proper documentation for the benchmark module.](https://issues.apache.org/jira/browse/SOLR-15625)

### CODE REVIEW

**Comment by epugh** [comment]
This looks like a great doc improvement, and we need to make benchmarking easier to approach.   @gerlowskija  I know you have actually used it...   If you were to LGTM this PR, I'd be happy to merge it etc.

**Comment by gerlowskija** [comment]
I'll take a look when I get a few moments @epugh 

But it's worth pinging @markrmiller to see if he might recall why this never got in - did it just fall through the cracks or was there some additional work needed?

File: solr/benchmark/README.md [hunk 1]
```diff
@@ -1,339 +1,423 @@
...
 
-`-jvmArgsAppend -DminiClusterBaseDir=/benchmark-data/mini-cluster`
+### 📚 Profilers
 
-In this case, new data will not be generated for the benchmark, even if you change parameters. The use case for this if
-you are running a query based benchmark and want to create a large index for testing and reuse (say hundreds of GB's).
-Be aware that with this system property set, that same mini-cluster will be reused for any benchmarks run, regardless of
-if that makes sense or not.
+- 📒 [docs/jmh-profilers.md](docs/docs/jmh-profilers.md)
+- 📒 [docs/jmh-profilers-setup.md](docs/docs/jmh-profilers-setup.md)
```
Comment (markrmiller) [inline solr/benchmark/README.md:423]: is docs/docs correct for these two links?

File: solr/benchmark/docs/jmh-profilers-setup.md [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+But you will also find an important _warning_ if you look closely at the logs.
+
+<br/>
+
+![](https://user-images.githubusercontent.com/448788/137613526-a188ff03-545c-465d-928d-bc433d2d204f.png)
+<span style="color: yellow; margin-left: 5px;">[WARN]</span> `Install JVM debug symbols to improve profile accuracy`
+
+<br/>
+
+We do not want **debug symbols** stripped from Java for the best experience.
```
Comment (markrmiller) [inline solr/benchmark/docs/jmh-profilers-setup.md:207]: maybe something a little more specific, like "... stripped from Java for optimal profiling accuracy and heap allocation analysis"

**Review by markrmiller — COMMENTED** [review]
Looks fine to me. I left a couple comments. I didn't even recall this PR existed.

File: solr/benchmark/README.md (hunk 1 above)
Comment (gerlowskija) [reply solr/benchmark/README.md]: It's not - I fixed the links in a recent patch.

File: solr/benchmark/docs/jmh-profilers-setup.md (hunk 2 above)
Comment (gerlowskija) [reply solr/benchmark/docs/jmh-profilers-setup.md]: Made some wording tweaks here to be more specific 👍

**Review by gerlowskija — APPROVED** [review]

**Comment by gerlowskija** [comment]
LGTM - I made a few tiny tweaks, but otherwise this is a clear net-positive for the benchmarking docs.

Thanks to:

- @markrmiller for the great docs
- @epugh for noticing that this had fallen through the cracks and resurrecting it.

Will merge later this morning.

**Pull request merged**